x/btcstaking/keeper/btc_delegations.go

```diff
@@ -227,11 +227,18 @@ func (k Keeper) setBTCDelegation(ctx context.Context, btcDel *types.BTCDelegatio
 
 // validateRestakedFPs ensures all finality providers are known to Babylon and at least
 // one of them is a Babylon finality provider. It also checks whether the BTC stake is
-// restaked to FPs of consumer chains
+// restaked to FPs of consumer chains. It enforces:
+// 1. Total number of FPs <= min of all consumers' max_multi_staked_fps
+// 2. At most 1 FP per consumer/BSN
 func (k Keeper) validateRestakedFPs(ctx context.Context, fpBTCPKs []bbn.BIP340PubKey) (bool, error) {
 	restakedToBabylon := false
 	restakedToConsumers := false
 
+	// Track FPs per consumer to enforce at-most-1-FP-per-consumer
+	consumerFPs := make(map[string]int)
```

Comment:
```suggestion
	// Track FPs per consumer to enforce at-most-1-FP-per-consumer
	consumerFPs := make(map[string]struct{})
```

to make "at most 1 FP per consumer" explicit